### PR TITLE
Add `chunksOf` function

### LINKED ISSUE #24: Add `chunksOf`
What do you think about adding the following function?

```haskell
chunksOf :: Int -> Slist a -> Slist (Slist a)
```

For ordinary lists it would work like this:

```haskell
ghci> chunksOf 3 [0..7]
[[0,1,2], [3,4,5], [6,7]]
```

But since we have access to the list length, we can implement a more efficient version of `chunksOf`.

### PR DESCRIPTION
**Pull request by mstruebing**
This doesn't already make it more efficient, I think this could be the next step to go.

I don't know if the runtime written there is correct.

closes #24

### CODE REVIEW

File: src/Slist.hs [hunk 1]
```diff
@@ -2065,3 +2067,12 @@ genericReplicate n x
...
 {-# INLINE genericReplicate #-}
+
+
+{- | @O(n)@. Splits the given slist into a slist of slists.
+
+> chunksOf 3 (slist [0..7]) == slist [slist [0,1,2], slist [3,4,5], slist [6,7]]
+
+-}
+chunksOf :: Int -> Slist a -> Slist (Slist a)
+chunksOf n xs = takeWhile (not.null) $ unfoldr (Just . splitAt n) xs
```
Comment (vrom911) [inline src/Slist.hs:2078]: I see two issues here, which we need to handle more carefully here:
* non-positive `n`s
* infinite lists

At the moment these two cases will result to problematic or incorrect situations :thinking:

**Review by vrom911 — COMMENTED** [review]
Hey @mstruebing!
Thanks for starting working on this. I wrote down a few potential issues I see with the function at the moment. You probably would need to reconsider the strategy of the implementation. Let me know if you have any questions! :slightly_smiling_face:

**Comment by vrom911** [comment]
Hey @mstruebing, are you planning to continue working on this?

**Comment by mstruebing** [comment]
Hey sorry, not right now, I have to much other things on my plate at the moment.

**Comment by vrom911** [comment]
No worries! I will close it for now, so some of us could take it when they have time.

**Pull request closed without merging**